hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ContainerReplicaPendingOps.java

```diff
@@ -261,6 +259,16 @@ private boolean completeOp(ContainerReplicaOp.PendingOpType opType,
     return found;
   }
 
+  public void registerOpCompletionHandler(
```

Comment:
Could we create a new Jira for these changes to ContainerReplicaPendingOps so it is a self contained change and not mixed into this larger one? We also need a couple of simple tests (or changes to existing tests) to ensure the handler is called in both scenarios.

It is so much easier to review a series of smaller Jira that a large one like this.

Reply:
sure, we can put all the changes of ContainerReplicaPendingOps into another separate jira